# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [4]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [13]:
air_bnb = pd.read_csv('files/AB_NYC_2019.csv')
air_bnb

id                                               name    host_id  \
0          2539                 Clean & quiet apt home by the park       2787   
1          2595                              Skylit Midtown Castle       2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !       4632   
3          3831                    Cozy Entire Floor of Brownstone       4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park       7192   
5          5099          Large Cozy 1 BR Apartment In Midtown East       7322   
6          5121                                    BlissArtsSpace!       7356   
7          5178                   Large Furnished Room Near B'way        8967   
8          5203                 Cozy Clean Guest Room - Family Apt       7490   
9          5238                 Cute & Cozy Lower East Side 1 bdrm       7549   
10         5295                   Beautiful 1br on Upper West Side       7702   
11         5441                    Central Manhattan/near Broadway       7989   
12         5803     Lovely Room 1, Garden, Best Area, Legal rental       9744   
13         6021   Wonderful Guest Bedroom in Manhattan for SINGLES      11528   
14         6090                      West Village Nest - Superhost      11975   
15         6848                   Only 2 stops to Manhattan studio      15991   
16         7097                  Perfect for Your Parents + Garden      17571   
17         7322                                    Chelsea Perfect      18946   
18         7726    Hip Historic Brownstone Apartment with Backyard      20950   
19         7750                  Huge 2 BR Upper East  Cental Park      17985   
20         7801                   Sweet and Spacious Brooklyn Loft      21207   
21         8024                     CBG CtyBGd HelpsHaiti rm#1:1-4      22486   
22         8025                           CBG Helps Haiti Room#2.5      22486   
23         8110                              CBG Helps Haiti Rm #2      22486   
24         8490             MAISON DES SIRENES1,bohemian apartment      25183   
25         8505                 Sunny Bedroom Across Prospect Park      25326   
26         8700  Magnifique Suite au N de Manhattan - vue Cloitres      26394   
27         9357                               Midtown Pied-a-terre      30193   
28         9518       SPACIOUS, LOVELY FURNISHED MANHATTAN BEDROOM      31374   
29         9657                   Modern 1 BR / NYC / EAST VILLAGE      21904   
...         ...                                                ...        ...   
48865  36472171                      1 bedroom in sunlit apartment   99144947   
48866  36472710                                 CozyHideAway Suite  274225617   
48867  36473044    The place you were dreaming for.(only for guys)  261338177   
48868  36473253                       Heaven for you(only for guy)  261338177   
48869  36474023                        Cozy, Sunny Brooklyn Escape    1550580   
48870  36474911      Cozy, clean Williamsburg 1- bedroom apartment    1273444   
48871  36475746      A LARGE ROOM - 1 MONTH MINIMUM - WASHER&DRYER  144008701   
48872  36476675                                      Nycity-MyHome    8636072   
48873  36477307                                  Brooklyn paradise  241945355   
48874  36477588                   Short Term Rental in East Harlem  214535893   
48875  36478343                              Welcome all as family  274273284   
48876  36478357    Cozy, Air-Conditioned Private Bedroom in Harlem  177932088   
48877  36479230             Studio sized room with beautiful light   65767720   
48878  36479723                                      Room for rest   41326856   
48879  36480292  Gorgeous 1.5 Bdr with a private yard- Williams...     540335   
48880  36481315  The Raccoon Artist Studio in Williamsburg New ...  208514239   
48881  36481615                   Peaceful space in Greenpoint, BK  274298453   
48882  36482231                          Bus

In [230]:
# How many neighborhood groups are available and which shows up the most?
#air_bnb.groupby('neighbourhood_group').head(5)
air_bnb.groupby('neighbourhood_group').count().mode()
# Manhattan

id   name  host_id  host_name  neighbourhood  latitude  longitude  \
0    373    373      373        373            373       373        373   
1   1091   1090     1091       1090           1091      1091       1091   
2   5666   5666     5666       5664           5666      5666       5666   
3  20104  20098    20104      20095          20104     20104      20104   
4  21661  21652    21661      21652          21661     21661      21661   

   room_type  price  minimum_nights  number_of_reviews  last_review  \
0        373    373             373                373          314   
1       1091   1091            1091               1091          876   
2       5666   5666            5666               5666         4574   
3      20104  20104           20104              20104        16447   
4      21661  21661           21661              21661        16632   

   reviews_per_month  calculated_host_listings_count  availability_365  
0                314                             373               373  
1                876                            1091              1091  
2               4574                            5666              5666  
3              16447                           20104             20104  
4              16632                           21661             21661

In [188]:
# Are private rooms the most popular in manhattan?
air_bnb.groupby('neighbourhood_group')['room_type'].head()
# No!

KeyError: 'Column not found: Private room'

In [206]:
# Which hosts are the busiest and based on their reviews?
air_bnb.groupby('host_id')['number_of_reviews'].count().nlargest(5)
# All of these have 0 nights available

host_id
219517861    327
107434423    232
30283594     121
137358866    103
12243051      96
Name: number_of_reviews, dtype: int64

In [74]:
#Which neighorhood group has the highest average price?
air_bnb.groupby('neighbourhood_group')['price'].mean().nlargest()

neighbourhood_group
Manhattan        196.875814
Brooklyn         124.383207
Staten Island    114.812332
Queens            99.517649
Bronx             87.496792
Name: price, dtype: float64

In [75]:
# Which neighbor hood group has the highest total price?
air_bnb.groupby('neighbourhood_group')['price'].sum().nlargest()

neighbourhood_group
Manhattan        4264527
Brooklyn         2500600
Queens            563867
Bronx              95459
Staten Island      42825
Name: price, dtype: int64

In [109]:
#Which top 5 hosts have the highest total price?
# air_bnb.groupby('host_name')['reviews_per_month'].sum().nlargest()
# air_bnb.groupby(['host_name', 'price'])['reviews_per_month'].sum().nlargest()
air_bnb.groupby(['host_name', 'price'])['reviews_per_month'].sum().nlargest()

host_name       price
Row NYC         100      76.18
David           50       52.54
Danielle        48       46.61
Alex And Zeena  85       39.49
Sonder (NYC)    229      34.86
Name: reviews_per_month, dtype: float64

In [216]:
# Who currently has no (zero) availability with a review count of 100 or more?

# air_bnb.query(['number_of_reviews > 100','host_id'])['availability_365'].head()
air_bnb.query('number_of_reviews > 100')['availability_365'].head()

3     194
7     220
8       0
9     188
11     39
Name: availability_365, dtype: int64

In [111]:
# What host has the highest total of prices and where are they located?

air_bnb.groupby(['host_name', 'neighbourhood'])['price'].sum().nlargest(1)

host_name     neighbourhood     
Sonder (NYC)  Financial District    57738
Name: price, dtype: int64

In [223]:
# When did Danielle from Queens last receive a review?


air_bnb.groupby('host_name','')['last_review'].head()


0        2018-10-19
1        2019-05-21
2               NaN
3        2019-07-05
4        2018-11-19
5        2019-06-22
6        2017-10-05
7        2019-06-24
8        2017-07-21
9        2019-06-09
10       2019-06-22
11       2019-06-23
12       2019-06-24
13       2019-07-05
14       2018-10-31
15       2019-06-29
16       2019-06-28
17       2019-07-01
18       2019-06-22
19              NaN
20       2011-12-28
21       2019-07-01
22       2019-01-01
23       2019-07-02
24       2019-06-19
25       2019-06-23
26              NaN
27       2017-08-13
28       2019-06-15
29       2019-04-19
            ...    
48808           NaN
48810           NaN
48813           NaN
48816           NaN
48818           NaN
48819           NaN
48821           NaN
48824           NaN
48825           NaN
48831           NaN
48832           NaN
48833           NaN
48836           NaN
48840           NaN
48842           NaN
48848           NaN
48849           NaN
48850           NaN
48858           NaN


## Further Questions

1. Which host has the most listings?

In [160]:
air_bnb.nlargest(1, 'calculated_host_listings_count')

id                                           name    host_id  \
38293  30181691  Sonder | 180 Water | Incredible 2BR + Rooftop  219517861   

          host_name neighbourhood_group       neighbourhood  latitude  \
38293  Sonder (NYC)           Manhattan  Financial District  40.70637   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
38293  -74.00645  Entire home/apt    302              29                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
38293         NaN                NaN                             327   

       availability_365  
38293               309

2. How many listings have completely open availability?

In [176]:
air_bnb.query('availability_365 < 1')['id'].count()

17533

3. What room_types have the highest review numbers?

In [177]:
air_bnb.groupby('room_type')['number_of_reviews'].sum().nlargest()

room_type
Entire home/apt    580403
Private room       538346
Shared room         19256
Name: number_of_reviews, dtype: int64

# Final Conclusion
1) How many neighborhood groups are available and which shows up the most? There are 5 neighborhood groups and Manhattan is the most popular
2)
3) Based on number of reviews, the busiest host is host_id: 219517861
4) The neighbourhood group that had the highest average price was Manhattan at $196.875814
5) The neighbourhood group that had the highest total price was again Manhattan, at $4,264,527
6) The top 5 hosts with the highest prices were Row NYC at a price of $100 and he had 76.18 reviews per month. Second was David at $50 with 52.54 reviews a month, Third was Danielle at a price of $48 at 46.61 reviews a month. Fourth was Alex and Zeena at $85 with only 39.49 reviews per month. The fifth highest was Sonder(NYC) at $229 with only 34.86 reviews per month
7) Host #8 has no availability and a review count of over 100
8) The host with the highest total prices was Sonder(NYC) in Financial District with a total price of $57738
9)

Further Questions:
1) Sonder (NYC) has the most listings
2) 17533 listings have complete availability
3) Entire home/apt have the highest review numbers at 580,403, while private room is second at 538,346. Third is a shared room with only 19,256 reviews
